src/ng/directive/ngOptions.js

```diff
@@ -618,10 +618,19 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
         var emptyOption_ = emptyOption && emptyOption[0];
         var unknownOption_ = unknownOption && unknownOption[0];
 
+        // If the compiled empty option was replaced by a comment because
+        // it had an "element" transclusion directive on it (such as ngIf)
+        // then compare against the pre-compiled empty element instead
+        if (emptyOption && emptyOption_.nodeType === NODE_TYPE_COMMENT) {
```

Comment:
We can actually remove this whole block then, can't we? (the tests pass without it)